base_layer/wallet/src/storage/sqlite_db/wallet.rs

```diff
@@ -535,7 +530,7 @@ impl WalletBackend for WalletSqliteDatabase {
         }
 
         // Now that all the decryption has been completed we can safely remove the cipher fully
-        let _ = (*current_cipher).take();
+        std::mem::drop((*current_cipher).take());
```

Comment:
nit: IMO this is the most idiomatic/simplest way to clear `current_cipher`.
```suggestion
        *current_cipher  = None;
```